src/cmd/descriptor.rs

```diff
@@ -45,3 +55,47 @@ impl Generate {
         Ok(())
     }
 }
+
+#[derive(Debug, clap::Args)]
+pub struct CountEdges {
+    /// The input descriptor file to generate signing bytes for
+    #[arg(default_value = "descriptor.json")]
+    input: PathBuf,
+    /// The file to write the resulting signing bytes to
+    #[arg(default_value = "edgecount.csv")]
+    output: PathBuf,
+}
+
+impl CountEdges {
+    pub fn run(&self) -> Result {
+        let descriptor = Descriptor::from_json(&self.input)?;
+        let mut counts: HashMap<PublicKey, i32> = HashMap::new();
```

Comment:
You can probably make this more efficient by `HashMap<&PublicKey, i32>`

Reply:
Fixed in [32234a7](https://github.com/helium/xorf-generator/pull/27/commits/32234a7a7679393f49f58c8910396f4780ec95f4)